File: Fundamentals-of-Programing141-main/in-class/19_loop_selection/loop_conversion.py
"""
In-Class Exercises:  Conversion between types of loops

File Name: loop_conversion.py
Course:    CPTR 141
"""

# Convert the following for loop to a while loop
count = 0
product = 1
for i in range(1, 5):
    product *= i
    count += 1
print("product: {}, count: {}".format(product, count))

# Convert the following while loop to a for loop
numbers = [10, 28, 2020]
i = 0
total = 0
while i < len(numbers):
    total += numbers[i]
    i += 1
average = total / len(numbers)
print("average: {}, total: {}".format(average, total))

# Convert the following while loop to a for loop
total = 0
x = 100
while x >= 75:
    total += x
    x -= 1
print("x: {}, total: {}".format(x, total))

File: Fundamentals-of-Programing142-main/in-class/14_cpp_looping/loop_conversion.cpp
/*************************************************************************
 *
 * In-Class Exercises:  Conversion between types of loops
 *
 * File Name: loop_conversion.cpp
 * Course:    CPTR 142
 */

#include <iostream> // for std::cout and std::cin streams

int main() {

  // Convert the following for loop to a while loop
  int count = 0, product = 1, i =1;
  while( i <= 4) {
    product *= i;
    count++;
    i++;
  }
  std::cout << "product: " << product << ", count: " << count << std::endl;

  // Convert the following while loop to a do-while loop
  int a = 0, sum = 0;
  do {
    sum += a;
    a--;
  }while (a != -1000);
  std::cout << "a: " << a << ", sum: " << sum << std::endl;

  // Convert the following while loop to a for loop
  int x= 100, total = 0;
  for (x=100;x >= 75; x--) {
    total += x;
   
  }
  std::cout << "x: " << x << ", total: " << total << std::endl << std::endl;

  return 0;
}